File: exams/bonifaz/unit2/Exam2/ExamUnit2/src/main/java/utils/MongoClientSingleton.java
package utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
/**
 *
 * @autor Christian Bonifaz, Code Creators, DCCO-ESPE
 */
public class MongoClientSingleton {
    private static final String URI = "mongodb+srv://bonifaz:bonifaz@cluster0.vleslwo.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
    private static MongoClient mongoClient = null;

    private MongoClientSingleton() {
       
    }

    public static MongoClient getInstance() {
        if (mongoClient == null) {
            synchronized (MongoClientSingleton.class) {
                if (mongoClient == null) {
                    mongoClient = MongoClients.create(URI);
                }
            }
        }
        return mongoClient;
    }
}

File: ws/bonifaz/u3/WS36ActionCRUD/ws36/utils/mongoClientSingleton.js
// utils/mongoClientSingleton.js
const mongoose = require('mongoose');

class MongoClientSingleton {
    constructor() {
        const uri = 'mongodb+srv://bonifaz:bonifaz@cluster0.vleslwo.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0';
        this.connection = mongoose.connect(uri, {
            useNewUrlParser: true,
            useUnifiedTopology: true,
        });
    }

    getConnection() {
        return this.connection;
    }
}

module.exports = new MongoClientSingleton();
